### PR TITLE
Replaced Mono.Nat with Open.NAT

### PR DESCRIPTION
**Pull request by Mailaender**
as suggested in https://github.com/OpenRA/OpenRA/issues/7140. I found a way to get us async support without raising the .NET version requirement.

~~I have some trouble getting this to work. See https://github.com/lontivero/Open.NAT/issues/20. This is the first time I experiment with AsyncTasks so I may be doing this wrong or Open.NAT is not yet ready for productive use with FRITZ!Box (a popular router in Germany).~~ Fixed.

### CODE REVIEW

**Comment by Mailaender** [comment]
PS: I just noticed that `msbuild` checks for a hard-coded .NET version number inside the DLL. Don't know how to workaround that.

**Comment by Phrohdoh** [comment]
Would an issue at https://github.com/Microsoft/msbuild possibly help with that limitation?

**Comment by Mailaender** [comment]
No, that is probably expected behavior although not helpful when working with AsyncBridge on older .NET versions.

**Comment by lontivero** [comment]
It seems there is a defect on Open.NAT with Fritz!Box. I have a fix candidate if you can test it.  Open.NAT [FritzRouter branch](https://github.com/lontivero/Open.NAT/tree/FritzRouter)

File: OpenRA.Game/Network/UPnP.cs [hunk 1]
```diff
@@ -9,112 +9,54 @@
...
-				Game.Settings.Server.NatDeviceAvailable = false;
-				Game.Settings.Server.AllowPortForward = false;
-			}
-		}
+		static NatDiscoverer natDiscoverer;
+		static Mapping mapping;
+		public static IPAddress ExternalIP;
 
-		public static void DeviceFound(object sender, DeviceEventArgs args)
+		public static async Task NatDiscovery()
```
Comment (ScottNZ) [inline OpenRA.Game/Network/UPnP.cs]: async/await are C# 5/.NET 4.5 features, which we don't support.

**Comment by Mailaender** [comment]
Now that we got that under control, could you ship a version on NuGet that does not have a strict requirement of .NET 4.5 which makes msbuild complain, but builds against the .NET 4.0 profile and uses AsyncBridge to supplement (in a separate `net40` folder)?

**Comment by Mailaender** [comment]
Updated to https://www.nuget.org/packages/Open.Nat/2.0.11 yet the .NET 4.0 compatibility issue is not yet formally resolved.

**Comment by lontivero** [comment]
I cannot use AsyncBridge to create a Open.NAT version that use .NET Frameworl 4.0 because it doesn't use only await and async keywords but also uses classes and methods aonly available for .NET Framework 4.5. 

I will continue fixing defects and providing support for Open.NAT however, even when it would love to have Open.NAT used in a great project like OpenRA, I won't rewrite parts of it.

**Comment by Mailaender** [comment]
Oh I wonder why I got this to work. Sounds dangerous though. I guess it could crash if I hit a method that requires a more recent .NET framework. Officially giving up here.

To be fair, we are not the only ones who can't simply switch to the latest and greatest .NET version. http://www.codeproject.com/Messages/4917441/Net.aspx

**Comment by rapgro** [comment]
Open.NAT was ported to .Net 3.5
https://github.com/lontivero/Open.NAT/pull/31

**Comment by Mailaender** [comment]
Superseded by https://github.com/OpenRA/OpenRA/pull/10281.

**Pull request closed without merging**